components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -2106,4 +2272,173 @@ private void handleInternalTokenRevocation(String consumerKey, Properties proper
+    }
+
+    /**
+     * FAPI validation to restrict the encryption algorithms.
+     * Link - https://openid.net/specs/openid-financial-api-part-2-1_0.html#encryption-algorithm-considerations
+     * @param encryptionAlgorithm
+     * @throws IdentityOAuthClientException
+     */
+    private void filterAllowedFAPIEncryptionAlgorithms(String encryptionAlgorithm)
+            throws IdentityOAuthClientException {
+
+        if (encryptionAlgorithm.equals(OAuthConstants.RESTRICTED_ENCRYPTION_ALGORITHM)) {
+            throw handleClientError(INVALID_REQUEST, "Invalid encryption algorithm requested");
+        }
+    }
+
+
+    /**
+     * If there are multiple hostnames in the registered redirect_uris,
+     * the Client MUST register a sector_identifier_uri.
+     * https://openid.net/specs/openid-connect-core-1_0.html#PairwiseAlg
+     * @param redirectURIs list of callback urls sent in the request
+     * @throws IdentityOAuthClientException
+     */
+    private void validateRedirectURIForPPID(List<String> redirectURIs, String sectorIdentifierURI)
+            throws IdentityOAuthClientException {
+
+        if (StringUtils.isEmpty(sectorIdentifierURI)) {
+            if (redirectURIs.size() > 1) {
+                String hostname = URI.create(redirectURIs.get(0)).getHost();
+                for (String redirectURI : redirectURIs) {
+                    URI uri = URI.create(redirectURI);
+                    if (!uri.getHost().equals(hostname)) {
+                        throw handleClientError(INVALID_REQUEST,
+                                "Sector identifier URI is needed for PPID calculation");
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * The value of the sector_identifier_uri MUST be a URL using the https scheme
+     * The values of the registered redirect_uris MUST be included in the elements of the array.
+     * https://openid.net/specs/openid-connect-core-1_0.html#PairwiseAlg
+     *
+     * @param sectorIdentifierURI sector identifier URI
+     * @param redirectURIs        callBack URLs
+     * @throws IdentityOAuthClientException
+     */
+    private void validateSectorIdentifierURI(String sectorIdentifierURI, List<String> redirectURIs) throws
+            IdentityOAuthClientException {
+
+        if (StringUtils.isBlank(sectorIdentifierURI)) {
+            return;
+        }
+        URI uri = URI.create(sectorIdentifierURI);
+        String scheme = uri.getScheme();
+        if (StringUtils.isBlank(scheme) || !scheme.equals(String.valueOf(HttpsURL.DEFAULT_SCHEME))) {
+            throw handleClientError(INVALID_REQUEST, "Invalid sector identifier URI");
+        }
+        //validate whether sectorIdentifierURI points to JSON file containing an array of redirect_uri values
+        String validateSectorIdentifierURI = IdentityUtil.getProperty(
+                OAuthConstants.VALIDATE_SECTOR_IDENTIFIER);
+        if (StringUtils.isNotEmpty(validateSectorIdentifierURI) &&
```

Comment:
StringUtils.isNotEmpty(validateSectorIdentifierURI) is not required as  Boolean.parseBoolean is null safe

Reply:
fixed

Reply:
seems this is not fixed yet

Reply:
fixed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2178/commits/ed3a3d57ecb1498f33c0b30ab2adf495a1f5706f